src/block_helpers.rs

```diff
@@ -168,6 +168,33 @@ pub fn slearn2<'a>(
     pb.observations[0]
 }
 
+pub fn ssetup_cache2<'a>(
```

Comment:
this name seems sub-optimal

Reply:
Indeed, but consistent with what we have right now. Didn't want to also rename those in an optimal way as it would be too many changes. Better to have everything consistent than something in-between.